irods_http_client/user_group_operations.py

```diff
@@ -0,0 +1,809 @@
+            raise TypeError('user_type must be a string')
+        if ((user_type != 'rodsuser') and (user_type != 'groupadmin') and (user_type != 'rodsadmin') and (user_type != '')):
+            raise ValueError('user_type must be set to rodsuser, groupadmin, or rodsadmin.')
+
+
+        headers = {
+            'Authorization': 'Bearer ' + self.token,
+            'Content-Type': 'application/x-www-form-urlencoded'
+        }
+
+        data = {
+            'op': 'set_user_type',
+            'name': name,
+            'zone': zone,
+            'user-type': user_type
+        }
+
+        r = requests.post(self.url_base + '/users-groups', headers=headers, data=data)
+
+        if (r.status_code / 100 == 2):
+            rdict = r.json()
+            if rdict['irods_response']['status_code']:
+                print('Failed to change type for user \'' + name + '\': iRODS Status Code' + str(rdict['irods_response']['status_code']))
+            else:
+                print('Type for user \'' + name + '\' changed successfully')
+            
+            return(
+                {
+                    'status_code': r.status_code,
+                    'data': rdict
+                }
+            )
+        else:
+            irods_err = ''
+            rdict = None
+            if (r.text != ''):
+                rdict = r.json()
+                irods_err = ': iRODS Status Code' + str(rdict['irods_response'])
+            print(f'Error <{r.status_code}>{irods_err}')
+
+            return(
+                {
+                    'status_code': r.status_code,
+                    'data': rdict
+                }
+            )
+    
+
+    def create_group(self, name: str):
+        """
+        Creates a new group. Requires rodsadmin or groupadmin privileges.
+        
+        Parameters
+        - name: The name of the group to be created.
+
+        Returns
+        - A dict containing the HTTP status code and iRODS response.
+        - The iRODS response is only valid if no error occurred during HTTP communication.
+        """
+        if (self.token == None):
+            raise RuntimeError('No token set. Use setToken() to set the auth token to be used')
+        if (not isinstance(name, str)):
+            raise TypeError('name must be a string')
+
+        headers = {
+            'Authorization': 'Bearer ' + self.token,
+            'Content-Type': 'application/x-www-form-urlencoded'
+        }
+
+        data = {
+            'op': 'create_group',
+            'name': name
+        }
+
+        r = requests.post(self.url_base + '/users-groups', headers=headers, data=data)
+
+        if (r.status_code / 100 == 2):
+            rdict = r.json()
+            if rdict['irods_response']['status_code']:
+                print('Failed to create group \'' + name + '\': iRODS Status Code' + str(rdict['irods_response']['status_code']))
+            else:
+                print('Group \'' + name + '\' created successfully')
```

Comment:
Remove all instances of lines like these.

_This applies to all endpoints in this PR._

Reply:
Issue #30 was opened to address the print statements. I can go ahead and start on replacing/removing these lines, whichever you feel would be better.

Reply:
No need. We'll address the print statements in a separate PR.

This is fine for now.